src/System.CommandLine.Subsystems.Tests/ErrorReportingSubsystemTests.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using FluentAssertions;
+using Xunit;
+using System.CommandLine.Parsing;
+
+namespace System.CommandLine.Subsystems.Tests;
+
+public class ErrorReportingSubsystemTests
+{
+    [Fact]
+    public void Report_when_single_error_writes_to_console_hack()
+    {
+        var error = new ParseError("a sweet error message");
+        var errors = new List<ParseError> { error };
+        var errorSubsystem = new ErrorReportingSubsystem();
+        var consoleHack = new ConsoleHack().RedirectToBuffer(true);
+
+        errorSubsystem.Report(consoleHack, errors);
+
+        consoleHack.GetBuffer().Trim().Should().Be(error.Message);
+    }
+
+    [Fact]
+    public void Report_when_multiple_error_writes_to_console_hack()
+    {
+        var error = new ParseError("a sweet error message");
+        var anotherError = new ParseError("another sweet error message");
+        var errors = new List<ParseError> { error, anotherError };
+        var errorSubsystem = new ErrorReportingSubsystem();
+        var consoleHack = new ConsoleHack().RedirectToBuffer(true);
+
+        errorSubsystem.Report(consoleHack, errors);
+
+        consoleHack.GetBuffer().Trim().Should().Be($"{error.Message}{Environment.NewLine}{anotherError.Message}");
+    }
+
+    [Fact]
+    public void Report_when_no_errors_writes_nothing_to_console_hack()
+    {
+        var errors = new List<ParseError> { };
+        var errorSubsystem = new ErrorReportingSubsystem();
+        var consoleHack = new ConsoleHack().RedirectToBuffer(true);
+
+        errorSubsystem.Report(consoleHack, errors);
+
+        consoleHack.GetBuffer().Trim().Should().Be("");
+    }
+
+    [Theory]
+    [InlineData("-v", false)]
+    [InlineData("-x", true)]
+    [InlineData("", false)]
+    public void GetIsActivated_tests(string input, bool result)
```

Comment:
```suggestion
    public void GetIsActivated_GivenParseResult_ShouldReturnExpectedResult(string input, bool result)
```
Tried coming up with a better name, but I would consider splitting this into two tests, one with expected of true, and one of the false cases.

I would argue this also makes naming the test way easier because the expected result is clear, and when a failure occurs, the expected result is more clear making the expectation more clear.

Reply:
Good call, definitely should be two tests 👍

Reply:
Pushed up a stab at that, lemme know what you think. The tests are repeated but the test name really clarifies intent and the not-necessarily-intuitive boolean behavior. For the error subsystem, GetIsActived should
- return true when there is an invalid input, or when there _is_ an error
- return false when there is a valid input, or when there _is not_ an error